catalog/ui/catalog-ui-search/src/main/webapp/component/sort-item/sort-item.view.js

```diff
@@ -36,45 +38,137 @@ define([
-            }).filter(function(type) {
+            }).filter(function (type) {
                 return blacklist.indexOf(type.id) === -1;
-            }).map(function(metacardType) {
+            }).map(function (metacardType) {
                 return {
                     label: metacardType.alias || metacardType.id,
                     value: metacardType.id
                 };
             });
 
-            this.sortAttribute.show(DropdownView.createSimpleDropdown({
-                hasFiltering: true,
-                list: sortAttributes,
-                defaultSelection: [this.model.get('attribute')]
+            sortAttributes.unshift({
+                label: 'Best Text Match',
+                value: 'RELEVANCE'
+            });
+
+            this.sortAttribute.show(new PropertyView({
+                model: new Property({
+                    enum: sortAttributes,
+
+                    value: [this.model.get('attribute')],
+                    id: 'Sort Attribute',
+                    enumFiltering: true
+                })
             }));
-            this.sortDirection.show(DropdownView.createSimpleDropdown({
-                list: [{
-                    label: 'Ascending',
-                    value: 'ascending'
-                }, {
-                    label: 'Descending',
-                    value: 'descending'
-                }],
-                defaultSelection: [this.model.get('direction')]
+            this.handleAttribute()
+
+            this.sortAttribute.currentView.$el.on('change', function (event) {
+                this.handleAttribute();
+            }.bind(this));
+        },
+        turnOffEditing: function () {
+            this.sortAttribute.currentView.turnOffEditing();
+            this.sortDirection.currentView.turnOffEditing();
+        },
+        turnOnEditing: function () {
+            this.sortAttribute.currentView.turnOnEditing();
+            if (!this.$el.hasClass('is-non-directional-sort')) {
+                this.sortDirection.currentView.turnOnEditing();
+            }
+        },
+        turnOnLimitedWidth: function () {
+            this.sortAttribute.currentView.turnOnLimitedWidth();
+            this.sortDirection.currentView.turnOnLimitedWidth();
+        },
+        handleAttribute: function () {
+            var attribute = this.sortAttribute.currentView.getCurrentValue()[0]; //this.model.get('attribute');
+            var ascendingLabel, descendingLabel;
+            if (metacardDefinitions.metacardTypes[attribute] === undefined) {
+                ascendingLabel = descendingLabel = '';
+                this.$el.toggleClass('is-non-directional-sort', true);
+            } else {
+                switch (metacardDefinitions.metacardTypes[attribute].type) {
+                    case 'DATE':
+                        ascendingLabel = 'Earliest';
+                        descendingLabel = 'Latest';
+                        break;
+                    case 'LOCATION':
```

Comment:
Wouldn't all geometries be closest/furthest, assuming we have a geospatial criteria... and otherwise not make sense at all?